tools/autograd/gen_variable_type.py

```diff
@@ -599,6 +608,16 @@
 """
 )
 
+ASSIGN_GRAD_FN_FOREACH = CodeTemplate(
+    """\
+for (const auto& i : c10::irange(self.size())) {
```

Comment:
this won't work for `_foreach_pow.ScalarAndTensor(Scalar self, Tensor[] exponent) -> Tensor[]`